lib/kafka_ex/stream.ex

```diff
@@ -0,0 +1,44 @@
+defmodule KafkaEx.Stream do
+  @moduledoc false
+  alias KafkaEx.Protocol.OffsetCommit.Request, as: OffsetCommitRequest
+  alias KafkaEx.Protocol.Fetch.Request, as: FetchRequest
+
+  defstruct worker_name: nil, fetch_request: %FetchRequest{}, consumer_group: nil
+
+  defimpl Enumerable do
+    def reduce(data, acc, fun) do
+      next_fun = fn offset ->
+        if data.fetch_request.auto_commit do
+          GenServer.call(data.worker_name, {
+            :offset_commit,
+            %OffsetCommitRequest{
+              consumer_group: data.consumer_group,
+              topic: data.fetch_request.topic,
+              partition: data.fetch_request.partition,
+              offset: offset, metadata: ""
+            }
+          })
+        end
+        response = data.worker_name
+        |> GenServer.call({:fetch, %{data.fetch_request| offset: offset}})
+        |> hd |> Map.get(:partitions) |> hd
```

Comment:
I'd prefer not to use `hd` so much - it's a bit opaque.  We can either pattern match `[element] = thing` or add helper functions to the appropriate modules, like `FetchResponse.first_response`.  Also I think we should try to stick to one pipe per line (I think credo would tell us this as well).